pkg/toml/util.go

```diff
@@ -0,0 +1,48 @@
+package toml
+
+import (
+	"os"
+	"fmt"
+	"github.com/naoina/toml"
+	"github.com/SeerUK/tid/pkg/types"
+)
+
+// BoltDatabaseFilename is the name of the database file name on disk.
+const TomlConfigFilename = "config.toml"
+
+// Open opens the configuration file or it creates it if it doesn't exist already.
+func Open(tidDir string) (types.TomlConfig, error) {
+	config := types.NewTomlConfig()
+
+	// Make the `path` if it does not exist.
+	err := os.MkdirAll(tidDir, os.ModePerm)
+	if err != nil {
+		return config, err
+	}
+
+	var configFilePath = fmt.Sprintf("%s/%s", tidDir, TomlConfigFilename)
+
+	f, err := os.OpenFile(configFilePath, os.O_RDONLY|os.O_CREATE, 0644)
+
+	if err != nil {
+		panic(err)
```

Comment:
What I was thinking was modifying it to return a pointer, so that nil can be returned alongside an error, rather than an empty struct. I suppose it depends on the use case though, if the defaults specified are fine then it should be fine to return the struct and an error, but I'm not sure how the errors are dealt with elsewhere and whether you want it to be something that stops the flow entirely?